### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by SquaredCanine**
Pl0x merge me

### CODE REVIEW

File: backend/core_transaction/controller/WorkflowTransactionRead.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+namespace backend\core\transaction\controller;
+
+
+use backend\core\common\lib\Workflow;
+use backend\core\common\exceptions\BackendUserException;
+use backend\core\transaction\model\TransactionModel;
+
+
```
Comment (JvanHengel) [inline backend/core_transaction/controller/WorkflowTransactionRead.php]: Okee, geen gezeik over de volgorde, maar plz wees dan wel consistent over de amount of newlines tussen namespace, import statements en class declaration *Autistic screeching*

File: backend/core_transaction/controller/WorkflowTransactionUpdate.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * @param TransactionModel $order_id
+     * @param $timestamp
+     * @return bool
+     * @throws \backend\core\common\exceptions\BackendException
+     */
+    private function updateTimestampIfNeeded(TransactionModel $order_id, $timestamp) : bool
+    {
+        if ($this->isTimestampUpdateNeeded($order_id, $timestamp)) {
```
Comment (JvanHengel) [inline backend/core_transaction/controller/WorkflowTransactionUpdate.php]: $order_id -> $orderId

File: backend/core_transaction/controller/WorkflowTransactionUpdate.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            $order_id->updateTimestamp($timestamp);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private function isTimestampUpdateNeeded(TransactionModel $order_id, $timestamp): bool
```
Comment (JvanHengel) [inline backend/core_transaction/controller/WorkflowTransactionUpdate.php]: Docblock

File: backend/core_transaction/lib/TransactionLib.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+namespace backend\core\transaction\lib;
+
+
+use backend\core\transaction\model\TransactionQuery;
+
+class TransactionLib
+{
+    public static function isOrderIDAvailable($order_id) : bool
```
Comment (JvanHengel) [inline backend/core_transaction/lib/TransactionLib.php]: Docblock

File: backend/core_transaction/model/TransactionModel.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    /**
+     * @param $order_id
+     * @param $timestamp
+     * @param $status
+     * @return TransactionModel
+     * @throws \backend\core\common\exceptions\BackendException
+     */
+    public static function create($order_id, $timestamp, $status) : TransactionModel
```
Comment (JvanHengel) [inline backend/core_transaction/model/TransactionModel.php]: Plz no space between ) and :

File: backend/core_transaction/controller/WorkflowTransactionUpdate.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        }
+    }
+
+    /**
+     * @param TransactionModel $order_id
+     * @param $status
+     * @return bool
+     * @throws \backend\core\common\exceptions\BackendException
+     */
+    private function updateStatusIfNeeded(TransactionModel $order_id, $status) : bool
```
Comment (JvanHengel) [inline backend/core_transaction/controller/WorkflowTransactionUpdate.php]: space

File: backend/core_transaction/model/TransactionQuery.php [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    /**
+     * @param $order_id
+     * @param $timestamp
+     * @param $status
+     * @return TransactionModel
+     * @throws BackendException
+     */
+    public static function create($order_id, $timestamp, $status) : TransactionModel
```
Comment (JvanHengel) [inline backend/core_transaction/model/TransactionQuery.php]: space

File: backend/core_transaction/model/TransactionQuery.php [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+        );
+
+        if ($result === false) {
+            throw new BackendException(self::ERROR_INSERTING_INTO_DATABASE);
+        } else {
+            return static::getFromDatabaseRow($order_id);
+        }
+    }
+
+    public static function updateTimestampByOrderID($timestamp, $order_id): bool
```
Comment (JvanHengel) [inline backend/core_transaction/model/TransactionQuery.php]: docblock

File: backend/core_transaction/model/TransactionQuery.php [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+        );
+
+        if ($result === false) {
+            throw new BackendException(self::ERROR_WHILE_UPDATING, [$order_id]);
+        } else {
+            return true;
+        }
+    }
+
+    public static function updateStatusByOrderID($status, $order_id): bool
```
Comment (JvanHengel) [inline backend/core_transaction/model/TransactionQuery.php]: docblock

File: composer.json [hunk 10]
```diff
@@ -21,6 +31,10 @@
             "backend\\core\\product\\lib\\": "backend/core_product/lib/",
             "backend\\core\\product\\model\\": "backend/core_product/model/",
             "backend\\core\\product\\view\\": "backend/core_product/view/",
+            "backend\\core\\transaction\\controller\\": "backend/core_transaction/controller/",
```
Comment (JvanHengel) [inline composer.json]: plz keep this file in alphabetical order <3

File: backend/core_transaction/controller/WorkflowTransactionRead.php (hunk 1 above)
Comment (SquaredCanine) [reply backend/core_transaction/controller/WorkflowTransactionRead.php]: Fixed

File: backend/core_transaction/controller/WorkflowTransactionUpdate.php (hunk 2 above)
Comment (SquaredCanine) [reply backend/core_transaction/controller/WorkflowTransactionUpdate.php]: On it!

Comment (SquaredCanine) [reply backend/core_transaction/controller/WorkflowTransactionUpdate.php]: Fixed it

File: backend/core_transaction/model/TransactionQuery.php (hunk 9 above)
Comment (SquaredCanine) [reply backend/core_transaction/model/TransactionQuery.php]: fixed

File: backend/core_transaction/model/TransactionQuery.php (hunk 8 above)
Comment (SquaredCanine) [reply backend/core_transaction/model/TransactionQuery.php]: fixed

File: backend/core_transaction/model/TransactionQuery.php (hunk 7 above)
Comment (SquaredCanine) [reply backend/core_transaction/model/TransactionQuery.php]: fixed

File: backend/core_transaction/controller/WorkflowTransactionUpdate.php (hunk 6 above)
Comment (SquaredCanine) [reply backend/core_transaction/controller/WorkflowTransactionUpdate.php]: fixed

File: backend/core_transaction/model/TransactionModel.php (hunk 5 above)
Comment (SquaredCanine) [reply backend/core_transaction/model/TransactionModel.php]: fixed

File: backend/core_transaction/lib/TransactionLib.php (hunk 4 above)
Comment (SquaredCanine) [reply backend/core_transaction/lib/TransactionLib.php]: fixed

File: backend/core_transaction/controller/WorkflowTransactionUpdate.php (hunk 3 above)
Comment (SquaredCanine) [reply backend/core_transaction/controller/WorkflowTransactionUpdate.php]: fixed

File: composer.json (hunk 10 above)
Comment (SquaredCanine) [reply composer.json]: Was ie niet eens voordat ik hem veranderde hahaha

Comment (SquaredCanine) [reply composer.json]: Fixed it tho

**Pull request merged**